### PR TITLE
PW-1762 Redirect shopper to external authorization

### PR DESCRIPTION
**Pull request by msilvagarcia**

### CODE REVIEW

File: adyen.php [hunk 1]
```diff
@@ -975,4 +910,109 @@ private function createCustomerMessage($message, CustomerThread $customerThread)
...
+    /**
+     * @return string
+     */
+    private function getStandardPaymentMethod()
+    {
+        $payments = '';
+        $this->context->smarty->assign(
+            array(
+                'locale' => $this->helper_data->getLocale($this->context->language),
+                // no locale in PrestaShop1.6 only languageCode that is en-en but we need en_EN
```
Comment (rikterbeek) [inline adyen.php]: should we define a default ?

Comment (msilvagarcia) [reply adyen.php]: what do you mean with this?

File: controllers/front/Payment.php [hunk 2]
```diff
@@ -412,4 +421,94 @@ public function buildRecurringData($request = array(), $payload)
...
+    }
+
+    private function processLocalPaymentMethod(\Cart $cart, $paymentType, $paymentIssuer)
+    {
+        $request = array();
+        $request = $this->buildBrowserData($request);
+        $request = $this->buildLocalPaymentMethodData($paymentType, $paymentIssuer, $request);
+        $request = $this->buildPaymentData($request);
+        $request = $this->buildMerchantAccountData($request);
+        $request = $this->buildRecurringData($request, $_REQUEST);
```
Comment (Aleffio) [inline controllers/front/Payment.php]: I think we don't need this one

File: controllers/front/Payment.php [hunk 3]
```diff
@@ -412,4 +421,94 @@ public function buildRecurringData($request = array(), $payload)
...
+                    [
+                        'message' => "There was an error with the payment method, please choose another one."
+                    ]
+                )
+            );
+            return;
+        }
+
+        switch ($response['resultCode']) {
+            case 'RedirectShopper':
```
Comment (Aleffio) [inline controllers/front/Payment.php]: Do we need a Default case to non-handle(error) the other flows? (I can only think of PresentToShopper at the moment)

Comment (msilvagarcia) [reply controllers/front/Payment.php]: I will add that

File: controllers/front/Payment.php (hunk 2 above)
Comment (msilvagarcia) [reply controllers/front/Payment.php]: We do. The `$request` array is just created there.

File: adyen.php (hunk 1 above)
Comment (rikterbeek) [reply adyen.php]: You are specifying that for 1.6 the format is incorrect en-en instaed of en_EN should we then not prefine it for 1.6 to en_EN or should we make a converter that makes en-en to en_EN ?

Comment (Aleffio) [reply adyen.php]: I think we need to do this in the getLocale function in the helper_data, I will create a youtrack ticket for this, so we can discuss

File: controllers/front/Payment.php (hunk 2 above)
Comment (Aleffio) [reply controllers/front/Payment.php]: buildRecurringData fills the request only if 'storeCc' is true, which is the checkbox we use for saving the credit cards, we don't need this for local payment methods. I am referring only to the buildRecurringData, the others are needed oc.

Comment (msilvagarcia) [reply controllers/front/Payment.php]:
```suggestion
```

**Pull request merged**